src/workerd/api/global-scope.h

```diff
@@ -382,6 +382,32 @@ struct ExportedHandler {
   jsg::Optional<jsg::Ref<ExecutionContext>> getCtx() {
     return ctx.map([&](jsg::Ref<ExecutionContext>& p) { return p.addRef(); });
   }
+
+ private:
+  template <typename T>
+  static jsg::LenientOptional<T> mapAddRef(jsg::Lock& js, jsg::LenientOptional<T>& function) {
+    return function.map([&](T& a) { return a.addRef(js); });
+  }
+
+ public:
```

Comment:
Please put all private stuff at the end of the struct/class, don't interleave private and public. Private stuff adds noise for people who are just trying to read the public interface, so it's best to keep it clearly separate.